pkg/azuredisk/controllerserver.go

```diff
@@ -225,6 +314,20 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 					},
 				},
 			}
+			// if migration monitoring is enabled and if target SKU is PremiumV2LRS, get the snapshot SKU
+			// the snapshot SKU is marked as source SKU
+			if skuName == armcompute.DiskStorageAccountTypesPremiumV2LRS && d.migrationMonitor != nil {
+				snapSku, err := d.getSnapshotSKU(ctx, sourceID)
```

Comment:
what about return the string directly? and move the warning logs into getSnapshotSKU func